Sources/TuistGenerator/Descriptors/SideEffectDescriptor.swift

```diff
@@ -0,0 +1,39 @@
+import Basic
+import Foundation
+import XcodeProj
+
+/// Generation Side Effect
+public enum SideEffect {
```

Comment:
What about naming this one `SideEffectDescriptor`. As the code base grows, suffixing the classes with their types will help find them in the code base.

Reply:
makes sense - also keeps things consistent 👍